scylla/src/transport/load_balancing/round_robin.rs

```diff
@@ -79,25 +80,27 @@ mod tests {
 
         let policy = RoundRobinPolicy::new();
 
-        let plans = (0..6)
+        let plans = (0..16)
             .map(|_| {
                 tests::get_plan_and_collect_node_identifiers(
                     &policy,
                     &tests::EMPTY_STATEMENT,
                     &cluster,
                 )
             })
-            .collect::<Vec<_>>();
+            .collect::<HashSet<_>>();
 
+        // Check if `plans` contains all possible round robin plans
         let expected_plans = vec![
             vec![1, 2, 3, 4, 5],
             vec![2, 3, 4, 5, 1],
             vec![3, 4, 5, 1, 2],
             vec![4, 5, 1, 2, 3],
             vec![5, 1, 2, 3, 4],
-            vec![1, 2, 3, 4, 5],
         ];
 
-        assert_eq!(plans, expected_plans);
+        for expected_plan in expected_plans.into_iter() {
+            assert!(plans.contains(&expected_plan));
+        }
```

Comment:
This just asserts that `expected_plans` is a subset of `plans`, it would be a bit better to check that both sets are equal.

I think you can easily change `expected_plans` to be a HashSet:

```rust
let expected_plans = HashSet::from([
    vec![1, 2, 3, 4, 5],
    /* ... */
]);
```

...and then bring back the old `assert_eq!`.